Assets/_Laz/Scripts/Environment/Puzzles/Connection/NodeConnectionBehaviour.cs

```diff
@@ -0,0 +1,102 @@
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(Collider))]
+    public class NodeConnectionBehaviour : MonoBehaviour
+    {
+        [SerializeField] private ParticleSystem _particleSystem = null;
+        [SerializeField] private Transform _meshTransform = null;
+        
+        private Node _node = null;
+        private LazoBehaviour _lazo = null;
+        
+        public void Initialize(Node node)
+        {
+            ClearParticleSystem();
+            _node = node;
+            _node.OnCanActivateChanged += HandleOnCanActivateChanged;
+            _node.OnNodeCompleted += HandleOnNodeCompleted;
+        }
+
+        public void CleanUp()
+        {
+            ClearParticleSystem();
+            HandleOnNodeCompleted();
+            _node.OnCanActivateChanged -= HandleOnCanActivateChanged;
+            _node.OnNodeCompleted += HandleOnNodeCompleted;
+            _node = null;
+        }
+
+        public void Reset()
+        {
+            HandleOnCanActivateChanged(false);
+        }
+
+        private void ClearParticleSystem()
+        {
+            _particleSystem.Stop();
+            _particleSystem.Clear();
+        }
+        
+        private void ActivateNodeIfNeeded(Collider other)
+        {
+            var lazo = other.GetComponent<LazoBehaviour>();
```

Comment:
Is it necessary to grab the Lazo during 'ontriggers' only to null it a few frames later? 
We could probably cache it no? Or can that result in problematic behavior?

Reply:
cuz anything could potentially be passed into the "OnTrigger" Zone. 

So i need to make sure the thing that passes through has a Lazo.

i'll move things around to make this method make a bit more sense

Reply:
Sounds good